File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.java
public class Solution {

    public int minSubArrayLen(int target, int[] nums) {
        int res = Integer.MAX_VALUE;
        int left = 0;
        int right = 0;
        int currSum = 0;

        while (right < nums.length) {
            currSum += nums[right];
            right++;

            while (currSum >= target) {
                res = Math.min(right - left, res);
                currSum -= nums[left];
                left++;
            }
        }

        return res == Integer.MAX_VALUE ? 0 : res;
    }
}

File: Data Structures & Algorithms/binary-search/submission-0.kt
class Solution {
    fun search(nums: IntArray, target: Int): Int {
        var left = 0
        var right = nums.size

        while (left <= right) {
            val mid = (left + right) / 2
            if (mid > nums.size - 1) return -1
            if (nums[mid] == target) return mid
            if (target < nums[mid]) {
                right = mid - 1
            } else {
                left = mid + 1
            }
        }

        return -1
    }
}
